app/src/androidTest/java/com/github/sdp/ratemyepfl/activity/ImageDetailActivityTest.kt

```diff
@@ -0,0 +1,60 @@
+package com.github.sdp.ratemyepfl.activity
+
+import android.content.Intent
+import androidx.test.core.app.ActivityScenario
+import androidx.test.core.app.ApplicationProvider
+import androidx.test.espresso.Espresso.onView
+import androidx.test.espresso.assertion.ViewAssertions.matches
+import androidx.test.espresso.matcher.ViewMatchers.isDisplayed
+import androidx.test.espresso.matcher.ViewMatchers.withId
+import androidx.test.ext.junit.rules.ActivityScenarioRule
+import com.github.sdp.ratemyepfl.R
+import com.github.sdp.ratemyepfl.utils.TestUtils.withDrawable
+import dagger.hilt.android.testing.HiltAndroidRule
+import dagger.hilt.android.testing.HiltAndroidTest
+import org.junit.After
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+
+
+@HiltAndroidTest
+class ImageDetailActivityTest {

```

Comment:
https://stackoverflow.com/questions/11523423/how-to-generate-zoom-pinch-gesture-for-testing-for-android, you can maybe use this to fully test the activity.

Reply:
Yes you're right. Thank you for the reference, it is really helpful.